gavel_ci/templates/console.html

```diff
@@ -0,0 +1,89 @@
+{% extends "layout.html" %}
+{% block body %}
+<h1>Gavel CI Build: <a href="{{url_for('jobserv.project', proj=proj)}}">{{proj}}</a> <a href="{{url_for('jobserv.build',proj=proj, build=build)}}">#{{build}}</a> - {{run}}</h1>
+<h3>
+    console.log
+    <span id='console-status'></span>
+    <span id='console-raw' style="display:none;">
+        (<a 
+          href="{{url_for('jobserv.run_artifact', proj=proj, build=build, run=run, p='console.log')}}"
+          download="{{proj}}-{{build}}-{{run}}-console.log">raw</a>)
+    </span>
+</h3>
+<div>
+    <pre
+        id="console-log-tail"
+        style="white-space: pre-wrap;"
+    ></pre>
+</div>
+{% endblock %}
+{% block script %}
+{{super()}}
+<script type="text/javascript">
+(function () {
+    let tailElement = $("#console-log-tail");
+    let downloadElement = $("#console-raw");
+    let statusElement = $("#console-status");
+    var offset = 0;
+    let messageTailWindow = 40;
+    var messageTail = [];
+    var status = '';
+    var interval = 1000; /* 1s */
```

Comment:
JobServ workers send log data to the server every 10s. So polling every second is probably a little much. There's no magic value I suppose, but this value means 90% of your calls aren't going to get new data. Maybe something like 3-5 seconds would be better.

Reply:
fixorized I'd forgot about that, ty